### PR TITLE
Feature/delete posts

### PR DESCRIPTION
**Pull request by belengg27**

### CODE REVIEW

File: app/src/androidTest/java/ch/proximeety/proximeety/data/repositories/UserRepositoryMockImplementation.kt [hunk 1]
```diff
@@ -173,6 +173,16 @@ class UserRepositoryMockImplementation : UserRepository {
         }
     }
 
+    override suspend fun deletePost(postId: String) {
+        posts.forEach { post ->
+            if (post.id == postId) {
+                posts.remove(post)
+            }
+        }
+
```
Comment (yanisdb) [inline app/src/androidTest/java/ch/proximeety/proximeety/data/repositories/UserRepositoryMockImplementation.kt:182]: Blank lines are not very useful

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/ProfileViewModel.kt [hunk 2]
```diff
@@ -80,8 +88,22 @@ class ProfileViewModel @Inject constructor(
...
+                }
+                _showDialog.value = false
+            }
         }
     }
 
+    fun onOpenDialogClicked() {
+        _showDialog.value = true
+    }
+
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/ProfileViewModel.kt:103]: It's better to use events then creating custom functions.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/ProfileViewModel.kt [hunk 3]
```diff
@@ -80,8 +88,22 @@ class ProfileViewModel @Inject constructor(
...
+
+    fun onCloseDialog() {
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/ProfileViewModel.kt:105]: It's better to use events then creating custom functions

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/FriendStat.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        Text(
+            text = count.toString(),
+            textAlign = TextAlign.Center,
+            fontSize = 25.sp,
+            modifier = Modifier.padding(top = padding, start = padding, end = padding)
+        )
+        Spacer(modifier = Modifier.height(5.dp))
+        Text(
+            text = "Friends",
+            fontSize = 20.sp,
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/FriendStat.kt:38]: Use the values for the MaterialTheme to keep things consistent.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/FriendStat.kt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .clip(CircleShape)
+            .border(width = 1.dp, color = Color.Black, shape = CircleShape)
+            .size(100.dp)
+    ) {
+        Text(
+            text = count.toString(),
+            textAlign = TextAlign.Center,
+            fontSize = 25.sp,
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/FriendStat.kt:32]: Use the values for the MaterialTheme to keep things consistent.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/FriendStat.kt [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+fun FriendStat(
+    count: Int
+) {
+    val padding = 7.dp
+    Column(
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .clip(CircleShape)
+            .border(width = 1.dp, color = Color.Black, shape = CircleShape)
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/FriendStat.kt:26]: Use the values for the MaterialTheme to keep things consistent.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/PostsStat.kt [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+@Composable
+fun PostsStat(
+    count: Int
+) {
+    val padding = 7.dp
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/PostsStat.kt:20]: Use the values for the MaterialTheme to keep things consistent.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/PostsStat.kt [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    Column(
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .clip(CircleShape)
+            .border(width = 1.dp, color = Color.Black, shape = CircleShape)
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/PostsStat.kt:26]: Use the values for the MaterialTheme to keep things consistent.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/PostsStat.kt [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+            .size(100.dp)
+    ) {
+        Text(
+            text = count.toString(),
+            textAlign = TextAlign.Center,
+            fontSize = 25.sp,
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/PostsStat.kt:32]: Use the values for the MaterialTheme to keep things consistent.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/PostsStat.kt [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+            modifier = Modifier.padding(top = padding, start = padding, end = padding)
+        )
+        Spacer(modifier = Modifier.height(5.dp))
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/PostsStat.kt:35]: Use the values for the MaterialTheme to keep things consistent.

File: app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/PostsStat.kt [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        Text(
+            text = "Posts",
+            fontSize = 20.sp,
```
Comment (yanisdb) [inline app/src/main/java/ch/proximeety/proximeety/presentation/views/profile/components/PostsStat.kt:38]: Use the values for the MaterialTheme to keep things consistent.

**Review by yanisdb — APPROVED** [review]
This is a good pull request. As minor comments, we can say that : 
 * There are a lot of hard-coded value, I couldn't add a comment on each.
 * for clarity, it's better to split composable in different files.
 *  the formatting is not perfect and there are empty lines.

**Review by Hamzezi — APPROVED** [review]
Good work. I left some minor comments that could be resolved at a later time, they do not impact the app in the eyes of the user, hence my approval.

**Pull request merged**